src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,972 @@
+                    {
+                        return (false, false);
+                    }
+                }
+
+                if (IsClassSpecificExpression(variable))
+                {
+                    containsClassSpecificStatement = true;
+                }
+            }
+
+            return (true, containsClassSpecificStatement);
+        }
+
+        /// <summary>
+        /// Creates new code actions for each introduce parameter possibility.
+        /// Does not create actions for overloads/trampoline if there are optional parameters or if the methodSymbol
+        /// is a constructor.
+        /// </summary>
+        private async Task<(ImmutableArray<CodeAction> actions, ImmutableArray<CodeAction> actionsAllOccurrences)?> GetActionsAsync(Document document,
+            TExpressionSyntax expression, IMethodSymbol methodSymbol, SyntaxNode containingMethod,
+            CancellationToken cancellationToken)
+        {
+            var (shouldDisplay, containsClassExpression) = await ShouldExpressionDisplayCodeActionAsync(
+                document, expression, cancellationToken).ConfigureAwait(false);
+            if (!shouldDisplay)
+            {
+                return null;
+            }
+
+            var actionsBuilder = ImmutableArray.CreateBuilder<CodeAction>();
+            var actionsBuilderAllOccurrences = ImmutableArray.CreateBuilder<CodeAction>();
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var singleLineExpression = syntaxFacts.ConvertToSingleLine(expression);
+            var nodeString = singleLineExpression.ToString();
+
+            if (!containsClassExpression)
+            {
+                actionsBuilder.Add(CreateNewCodeAction(FeaturesResources.and_update_call_sites_directly, allOccurrences: false, trampoline: false, overload: false));
+                actionsBuilderAllOccurrences.Add(CreateNewCodeAction(FeaturesResources.and_update_call_sites_directly, allOccurrences: true, trampoline: false, overload: false));
+            }
+
+            if (methodSymbol.MethodKind is not MethodKind.Constructor)
+            {
+                actionsBuilder.Add(CreateNewCodeAction(
+                    FeaturesResources.into_extracted_method_to_invoke_at_call_sites, allOccurrences: false, trampoline: true, overload: false));
+                actionsBuilderAllOccurrences.Add(CreateNewCodeAction(
+                    FeaturesResources.into_extracted_method_to_invoke_at_call_sites, allOccurrences: true, trampoline: true, overload: false));
+
+                actionsBuilder.Add(CreateNewCodeAction(
+                    string.Format(FeaturesResources.into_new_overload_of_0, nodeString), allOccurrences: false, trampoline: false, overload: true));
+                actionsBuilderAllOccurrences.Add(CreateNewCodeAction(
+                    string.Format(FeaturesResources.into_new_overload_of_0, nodeString), allOccurrences: true, trampoline: false, overload: true));
```

Comment:
how do you make an overload of a local function?  the language doesn't allow that.

Reply:
added check